### PR TITLE
2.1.0.rc1

### PR DESCRIPTION
**Pull request by omry**
Bump version to 2.1.0.rc1 and created the release notes.

I would appreciate a cursory read of the release notes to see it's readable and clear.
I am planning to release it later today.

This will be much easier to read using view file:
image

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,109 @@
...
+- Add DictConfig support for keys of type int, float and bool ([#149](https://github.com/omry/omegaconf/issues/149)), ([#483](https://github.com/omry/omegaconf/issues/483))
+- Structured Configs fields without a value are now automatically treated as `OmegaConf.MISSING` ([#390](https://github.com/omry/omegaconf/issues/390))
+- Add minimal support for typing.TypedDict ([#473](https://github.com/omry/omegaconf/issues/473))
+#### Interpolation and resolvers
+- Support for relative interpolation ([#48](https://github.com/omry/omegaconf/issues/48))
+- Add ability to nest interpolations, e.g. ${foo.${bar}}}, ${oc.env:{$var1},${var2}}, or ${${func}:x1,x2} ([#445](https://github.com/omry/omegaconf/issues/445))
+- Custom resolvers can now access the parent and the root config nodes ([#266](https://github.com/omry/omegaconf/issues/266))
+- Bracketed keys may be used as an alternative form to dot notation, e.g. foo.1 is equivalent to foo[1], [foo].1 and [foo][1].
+  This affects `OmegaConf.{update, select}` and interpolation usage. ([#179](https://github.com/omry/omegaconf/issues/179))
```
[the remark this answers is not in the document]
Comment (omry) [reply NEWS.md]: Sure.

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,109 @@
+## 2.1.0.rc1 (2021-05-12)
+This is the first release candidate of OmegaConf 2.1.
```
Comment (omry) [inline NEWS.md:2]:
```suggestion
```

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,108 @@
...
+- Structured Configs fields without a value are now automatically treated as `OmegaConf.MISSING` ([#390](https://github.com/omry/omegaconf/issues/390))
+- Add minimal support for typing.TypedDict ([#473](https://github.com/omry/omegaconf/issues/473))
+#### Interpolation and resolvers
+- Support for relative interpolation ([#48](https://github.com/omry/omegaconf/issues/48))
+- Add ability to nest interpolations, e.g. ${foo.${bar}}}, ${oc.env:{$var1},${var2}}, or ${${func}:x1,x2} ([#445](https://github.com/omry/omegaconf/issues/445))
+- Custom resolvers can now access the parent and the root config nodes ([#266](https://github.com/omry/omegaconf/issues/266))
+- For `OmegaConf.{update, select}` and in interpolations, bracketed keys may be used as an alternative form to dot notation,
+  e.g. foo.1 is equivalent to foo[1], [foo].1 and [foo][1]. ([#179](https://github.com/omry/omegaconf/issues/179))
+- The `env` resolver parses environment variables. Supported types includes primitives (int, float, bool, ...) and containers like dict and list. The used grammar is a subset of the interpolation grammar. ([#445](https://github.com/omry/omegaconf/issues/445))
```
Comment (odelalleau) [inline NEWS.md:24]: Remove that item?

File: NEWS.md [hunk 4]
```diff
@@ -1,3 +1,108 @@
...
+- OmegaConf.to_container() raises a ValueError on invalid input ([#418](https://github.com/omry/omegaconf/issues/418))
+- Fix ConfigKeyError in some cases when merging lists containing interpolation values ([#422](https://github.com/omry/omegaconf/issues/442))
+- DictConfig.get() in struct mode return None like standard Dict for non-existing keys ([#425](https://github.com/omry/omegaconf/issues/425))
+- Fix bug where interpolations were unnecessarily resolved during merge ([#431](https://github.com/omry/omegaconf/issues/431))
+- Fix bug where assignment of an invalid value to a ListConfig raised an exception but left the object modified. ([#433](https://github.com/omry/omegaconf/issues/433))
+- When initializing a Structured Config with an incorrectly-typed value, the resulting ValidationError now properly reports the offending value in its error message. ([#435](https://github.com/omry/omegaconf/issues/435))
+- Fix assignment of a Container to itself causing it to clear it's content ([#449](https://github.com/omry/omegaconf/issues/449))
```
Comment (odelalleau) [inline NEWS.md:55]:
```suggestion
- Fix assignment of a Container to itself causing it to clear its content ([#449](https://github.com/omry/omegaconf/issues/449))
```

File: NEWS.md [hunk 5]
```diff
@@ -1,3 +1,108 @@
...
+- OmegaConf.update() is now merging dict/list values into the destination node by default. Call with merge=False to replace instead. ([#667](https://github.com/omry/omegaconf/issues/667))
+- `register_resolver()` is deprecated in favor of `register_new_resolver()`, allowing resolvers to (i) take non-string arguments like int, float, dict, interpolations, etc. and (ii) control the cache behavior (now disabled by default) ([#426](https://github.com/omry/omegaconf/issues/426))
+- Merging a MISSING value onto an existing value no longer changes the target value to MISSING. ([#462](https://github.com/omry/omegaconf/issues/462))
+- When resolving an interpolation of a config value with a primitive type, the interpolated value is validated and possibly converted based on the node's type. ([#488](https://github.com/omry/omegaconf/issues/488))
+- DictConfig and ListConfig shallow copy is now performing a deepcopy ([#492](https://github.com/omry/omegaconf/issues/492))
+- A custom resolver interpolation whose output is a list or dictionary is now automatically converted into a ListConfig or DictConfig. ([#540](https://github.com/omry/omegaconf/issues/540))
```
Comment (odelalleau) [inline NEWS.md:85]: Was supposed to be removed?

File: NEWS.md [hunk 6]
```diff
@@ -1,3 +1,108 @@
...
+- `OmegaConf.select()`, `DictConfig.{get(),pop()}`, `ListConfig.{get(),pop()}` no longer return the specified default value when the accessed key is an interpolation that cannot be resolved: instead, an exception is raised. ([#543](https://github.com/omry/omegaconf/issues/543))
+- OmegaConf.to_container now takes a `structured_config_mode` keyword argument. Setting `structured_config_mode=SCMode.DICT_CONFIG` causes `to_container` to not convert Structured Config objects to python dicts (it leaves them as DictConfig objects). ([#548](https://github.com/omry/omegaconf/issues/548))
```
Comment (odelalleau) [inline NEWS.md:87]: Should this be in the API enhancements section?

**Review by odelalleau — CHANGES_REQUESTED** [review]
A few small comments

File: NEWS.md (hunk 5 above)
Comment (omry) [reply NEWS.md:85]: Yes, I'll remove it.

File: NEWS.md (hunk 3 above)
Comment (omry) [reply NEWS.md:24]: yes

File: NEWS.md (hunk 6 above)
Comment (omry) [reply NEWS.md:87]: yes.

**Pull request merged**